camel/agents/embodied_agent.py

```diff
@@ -169,20 +100,30 @@ def step(
 
         # NOTE: Only single output messages are supported
         output_message = output_messages[0]
-        explanation, code = self.get_explanation_and_code(
-            output_message.content)
-        if self.verbose:
-            print_text_animated(self.logger_color +
-                                f"> Explanation:\n{explanation}")
-            print_text_animated(self.logger_color + f"> Code:\n{code}")
+        explanations_list, codes_list =\
+            output_message.extract_text_and_code_prompts()
```

Comment:
Prefer to use parenthesis for formatting and simplify the naming  
```suggestion
        explanations, codes = (
            output_message.extract_text_and_code_prompts())
```